onnxruntime/core/providers/xnnpack/nn/max_pool.cc

```diff
@@ -54,6 +55,9 @@ bool MaxPool::IsOnnxNodeSupported(const NodeUnit& node_unit,
     // input of maxpool could be fp16/fp32/fp64,i8/u8 according to ONNX
     if (x_type == nullptr ||
         (x_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_FLOAT &&
+#ifdef XNNPACK_FP16_SUPPORTED
```

Comment:
Why the macro `XNNPACK_FP16_SUPPORTED` checked here?  The following code didn't wrap around by this macro.

Reply:
It can be remove. Though it will not be hit if XNNPACK_FP16_SUPPORTED is not defined